src/Interpreters/GraceHashJoin.cpp

```diff
@@ -281,6 +281,8 @@ GraceHashJoin::GraceHashJoin(
 
 void GraceHashJoin::initBuckets()
 {
+    std::lock_guard current_bucket_lock(current_bucket_mutex);
```

Comment:
Is this lock a must? Since initBuckets is only called by initialize which is guarded by lock.